website/content/en/preview/getting-started/getting-started-with-terraform/_index.md

```diff
@@ -108,42 +108,57 @@ module "vpc" {
   single_nat_gateway     = true
   one_nat_gateway_per_az = false
 
+  public_subnet_tags = {
+    "kubernetes.io/cluster/${local.cluster_name}" = "shared"
+    "kubernetes.io/role/elb"                      = 1
+  }
+
   private_subnet_tags = {
-    "kubernetes.io/cluster/${local.cluster_name}" = "owned"
-    # Tags subnets for Karpenter auto-discovery
-    "karpenter.sh/discovery" = local.cluster_name
+    "kubernetes.io/cluster/${local.cluster_name}" = "shared"
+    "kubernetes.io/role/internal-elb"             = 1
   }
 }
 
 module "eks" {
   # https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest
   source  = "terraform-aws-modules/eks/aws"
-  version = "18.17.0"
+  version = "18.20.5"
 
   cluster_name    = local.cluster_name
-  cluster_version = "1.21"
+  cluster_version = "1.22"
```

Comment:
If we can extract the HCL out of the README and put it into actual `*.tf` files somewhat similar to what was done for CloudFormation where the CloudFormation is injected via the template tags, that will give us 2 things:
1. Its easier to test and validate these changes because we can `cd` to the directory that holds the Terraform file and simply `terraform init && terraform apply`
2. Once the contents are back into Terraform files, we could enable Renovate which will track module updates and submit PRs to update https://github.com/DrFaust92/terraform-kubernetes-ebs-csi-driver/pull/78

In general though, the values are set statically to ensure that the guidance for users has been validated first. In theory, we can leave off line 128 and the underlying EKS module/service will pull the latest K8s version offered. However, this can be problematic if changes are required for supporting the latest version.

Reply:
> In general though, the values are set statically to ensure that the guidance for users has been validated first.

I agree this is definitely a good system. It's not easy to make sure it's always up to date and works, which gives value to hosting this in another repo like you've done. Was mostly curious if you had a good solution :)